### PR TITLE
🚮 Delete gulp test

### LINKED ISSUE #22507: Separate `gulp test` into `gulp unit` and `gulp integration`
Today, all of AMP's unit and integration tests are run by [build-system/tasks/runtime-test/index.js](https://github.com/ampproject/amphtml/blob/b1afbe2d15ca011fe662ff39a8bbfac7fd03bebb/build-system/tasks/runtime-test/index.js), which is 700+ lines long and growing. However, unit tests and integration tests use different sets of configs, flags, test files, and pre-requisite tasks, and serve different purposes. Let's split them into two tasks called `gulp unit` and `gulp integration`.

**Things to do:**

- [x] Move [code coverage report uploading](https://github.com/ampproject/amphtml/blob/b1afbe2d15ca011fe662ff39a8bbfac7fd03bebb/build-system/tasks/runtime-test/index.js#L561-L594) (Travis only) into a new `gulp codecov-upload` task that's run at the end of `Local Tests` (Done in #22515)
- [x] Move task-specific code from `tasks/runtime-test/index.js` into `tasks/unit.js` and `tasks/integration.js`
- [x] Move common code into `tasks/runtime-test/helpers.js`
- [ ] Simplify [`getConfig()`](https://github.com/ampproject/amphtml/blob/b1afbe2d15ca011fe662ff39a8bbfac7fd03bebb/build-system/tasks/runtime-test/index.js#L62-L145), [`printArgvMessages()`](https://github.com/ampproject/amphtml/blob/b1afbe2d15ca011fe662ff39a8bbfac7fd03bebb/build-system/tasks/runtime-test/index.js#L147-L240), and [`runTests()`](https://github.com/ampproject/amphtml/blob/b1afbe2d15ca011fe662ff39a8bbfac7fd03bebb/build-system/tasks/runtime-test/index.js#L242-L453)
- [x] Refactor `tasks/report-test-status.js` to understand the new gulp task structure
- [x] Update `TESTING.md`

/cc @ampproject/wg-infra

### PR DESCRIPTION
**Pull request by estherkim**
Closes #22507

### CODE REVIEW

File: build-system/pr-check/build-targets.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ const targetMatchers = [
     targets: ['INTEGRATION_TEST'],
     func: file => {
       return (
-        file.startsWith('build-system/tasks/runtime-test/') ||
+        file == 'build-system/tasks/integration.js' ||
```
Comment (rsimha) [inline build-system/pr-check/build-targets.js]: Maybe include the respective helper files too, so that tests are run if they are changed?

**Comment by danielrozenberg** [comment]
I still think we should wait with this a while, or at least keep a no-op `test` task that informs developers about the change. Not all AMP developers work on AMP continuously, some of them come back after a few months. Getting a "task not found" error for `gulp test` can be confusing and waste developer time

**Comment by rsimha** [comment]
> I still think we should wait with this a while, or at least keep a no-op `test` task that informs developers about the change. Not all AMP developers work on AMP continuously, some of them come back after a few months. Getting a "task not found" error for `gulp test` can be confusing and waste developer time

+1 for a no-op `test` task that redirects you to one of `unit`, `integration`, or `e2e`.

File: build-system/tasks/runtime-test/index.js [hunk 2]
```diff
@@ -700,32 +56,4 @@ module.exports = {
...
-  'files': '  Runs tests for specific files',
-  'nohelp': '  Silence help messages that are printed prior to test run',
-  'a4a': '  Runs all A4A tests',
-  'coverage': '  Run tests in code coverage mode',
-  'headless': '  Run tests in a headless Chrome window',
-  'local-changes':
-    '  Run unit tests directly affected by the files ' +
-    'changed in the local branch',
-};
+test.description = 'Runs tests (DEPRECIATED)';
```
Comment (danielrozenberg) [inline build-system/tasks/runtime-test/index.js]: DEPRECATED 🙃

Comment (estherkim) [reply build-system/tasks/runtime-test/index.js]: :see_no_evil: :see_no_evil: :see_no_evil: I meant to say that it lowered in value

Comment (rsimha) [reply build-system/tasks/runtime-test/index.js]: 😃

**Pull request merged**